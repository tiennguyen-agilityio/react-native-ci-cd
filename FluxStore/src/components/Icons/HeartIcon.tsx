import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

import {useThemeStore} from '@/stores';
import {IconProps} from '@/interfaces';
import {colors} from '@/themes';

export const HeartIcon = memo(
  ({
    width = 17,
    height = 15,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="heart-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 17 15" fill="none">
          <Path
            d="M8.504 14.6l-1.178-1.037C3.154 9.9.4 7.481.4 4.516a4.159 4.159 0 01.328-1.658c.221-.526.55-1.004.964-1.406.415-.402.908-.72 1.451-.935A4.536 4.536 0 014.854.2a4.984 4.984 0 012.009.436c.63.283 1.19.693 1.641 1.203.451-.51 1.01-.919 1.638-1.202A4.98 4.98 0 0112.145.2a4.536 4.536 0 011.712.317 4.425 4.425 0 011.45.935c.415.402.743.88.965 1.406a4.16 4.16 0 01.328 1.658c0 2.965-2.754 5.384-6.926 9.056L8.504 14.6z"
            fill={color ?? isActive ? colors.red[400] : icon.default}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
