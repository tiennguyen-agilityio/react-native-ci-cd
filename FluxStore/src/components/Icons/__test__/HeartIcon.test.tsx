import {render, fireEvent, screen} from '@testing-library/react-native';

import {HeartIcon} from '../HeartIcon';

describe('HeartIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<HeartIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<HeartIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<HeartIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('heart-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<HeartIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('heart-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
