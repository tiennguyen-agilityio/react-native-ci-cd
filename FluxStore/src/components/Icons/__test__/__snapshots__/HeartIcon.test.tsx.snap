// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeartIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="heart-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={17}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 17,
        },
      ]
    }
    vbHeight={15}
    vbWidth={17}
    width={17}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M8.504 14.6l-1.178-1.037C3.154 9.9.4 7.481.4 4.516a4.159 4.159 0 01.328-1.658c.221-.526.55-1.004.964-1.406.415-.402.908-.72 1.451-.935A4.536 4.536 0 014.854.2a4.984 4.984 0 012.009.436c.63.283 1.19.693 1.641 1.203.451-.51 1.01-.919 1.638-1.202A4.98 4.98 0 0112.145.2a4.536 4.536 0 011.712.317 4.425 4.425 0 011.45.935c.415.402.743.88.965 1.406a4.16 4.16 0 01.328 1.658c0 2.965-2.754 5.384-6.926 9.056L8.504 14.6z"
        fill={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`HeartIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="heart-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={15}
    bbWidth={17}
    fill="none"
    focusable={false}
    height={15}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 15,
          "width": 17,
        },
      ]
    }
    vbHeight={15}
    vbWidth={17}
    width={17}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M8.504 14.6l-1.178-1.037C3.154 9.9.4 7.481.4 4.516a4.159 4.159 0 01.328-1.658c.221-.526.55-1.004.964-1.406.415-.402.908-.72 1.451-.935A4.536 4.536 0 014.854.2a4.984 4.984 0 012.009.436c.63.283 1.19.693 1.641 1.203.451-.51 1.01-.919 1.638-1.202A4.98 4.98 0 0112.145.2a4.536 4.536 0 011.712.317 4.425 4.425 0 011.45.935c.415.402.743.88.965 1.406a4.16 4.16 0 01.328 1.658c0 2.965-2.754 5.384-6.926 9.056L8.504 14.6z"
        fill={
          {
            "payload": 4294930030,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
